From 7356edbd79758f7157a149ed43c59fd43c733160 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Tue, 10 Oct 2017 17:12:46 +0800
Subject: [PATCH 2618/5242] MLK-16563-3: ARM64: dts: support AUDIO IO board
 with imx8qxp-mek

commit  5b7537beaf98cbbc6a20a07f664c5f788dde861e from
https://source.codeaurora.org/external/imx/linux-imx.git

AUDIO IO board (SCH-501-1-005960-A1) has CS42888 codec and connect
to MCIMX-8X-BB via PCIe.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts |   73 ++++++++++++++++++++-
 1 file changed, 72 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index 9c51e74..6986f1d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -65,6 +65,15 @@
 			gpio = <&pca9557_b 2 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
+
+		reg_audio: fixedregulator@2 {
+			compatible = "regulator-fixed";
+			reg = <2>;
+			regulator-name = "cs42888_supply";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+		};
 	};
 
 	sound: sound {
@@ -109,6 +118,15 @@
 		dais = <&sai4>, <&sai5>;
 		amix-controller = <&amix>;
 	};
+
+	sound-cs42888 {
+		compatible = "fsl,imx8qm-sabreauto-cs42888",
+				"fsl,imx-audio-cs42888";
+		model = "imx-cs42888";
+		esai-controller = <&esai0>;
+		audio-codec = <&cs42888>;
+		status = "okay";
+	};
 };
 
 &acm {
@@ -124,6 +142,20 @@
 	status = "okay";
 };
 
+&esai0 {
+	compatible = "fsl,imx8qxp-v1-esai";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_esai0>;
+	assigned-clocks = <&clk IMX8QXP_ACM_ESAI0_MCLK_SEL>,
+			<&clk IMX8QXP_AUD_PLL0_DIV>,
+			<&clk IMX8QXP_AUD_ACM_AUD_PLL_CLK0_DIV>,
+			<&clk IMX8QXP_AUD_ESAI_0_EXTAL_IPG>;
+	assigned-clock-parents = <&clk IMX8QXP_AUD_ACM_AUD_PLL_CLK0_CLK>;
+	assigned-clock-rates = <0>, <786432000>, <49152000>, <49152000>;
+	dmas = <&edma2 23 0 3>, <&edma2 21 0 2>;
+	status = "okay";
+};
+
 &sai4 {
 	assigned-clocks = <&clk IMX8QXP_AUD_PLL0_DIV>,
 			<&clk IMX8QXP_AUD_ACM_AUD_PLL_CLK0_DIV>,
@@ -146,7 +178,29 @@
 
 &iomuxc {
 	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+
 	imx8qxp-mek {
+		pinctrl_hog: hoggrp {
+			fsl,pins = <
+				SC_P_MCLK_OUT0_ADMA_ACM_MCLK_OUT0	0x0600004c
+			>;
+		};
+
+		pinctrl_esai0: esai0grp {
+			fsl,pins = <
+				SC_P_ESAI0_FSR_ADMA_ESAI0_FSR		0xc600004c
+				SC_P_ESAI0_FST_ADMA_ESAI0_FST		0xc600004c
+				SC_P_ESAI0_SCKR_ADMA_ESAI0_SCKR		0xc600004c
+				SC_P_ESAI0_SCKT_ADMA_ESAI0_SCKT		0xc600004c
+				SC_P_ESAI0_TX0_ADMA_ESAI0_TX0		0xc600004c
+				SC_P_ESAI0_TX1_ADMA_ESAI0_TX1		0xc600004c
+				SC_P_ESAI0_TX2_RX3_ADMA_ESAI0_TX2_RX3	0xc600004c
+				SC_P_ESAI0_TX3_RX2_ADMA_ESAI0_TX3_RX2	0xc600004c
+				SC_P_ESAI0_TX4_RX1_ADMA_ESAI0_TX4_RX1	0xc600004c
+				SC_P_ESAI0_TX5_RX0_ADMA_ESAI0_TX5_RX0	0xc600004c
+			>;
+		};
 
 		pinctrl_lpuart0: lpuart0grp {
 			fsl,pins = <
@@ -230,7 +284,6 @@
 				SC_P_SAI1_RXFS_ADMA_SAI1_TXFS	0x0600004c
 				SC_P_SPI0_CS1_ADMA_SAI1_TXD	0x0600004c
 				SC_P_SPI2_CS0_LSIO_GPIO1_IO00	0x0600004c
-				SC_P_MCLK_OUT0_ADMA_ACM_MCLK_OUT0	0x0600004c
 			>;
 		};
 
@@ -409,6 +462,24 @@
 				<&clk IMX8QXP_AUD_MCLKOUT0>;
 		assigned-clock-rates = <786432000>, <12288000>, <12288000>;
 	};
+
+	cs42888: cs42888@48 {
+		compatible = "cirrus,cs42888";
+		reg = <0x48>;
+		clocks = <&clk IMX8QXP_AUD_MCLKOUT0>;
+		clock-names = "mclk";
+		VA-supply = <&reg_audio>;
+		VD-supply = <&reg_audio>;
+		VLS-supply = <&reg_audio>;
+		VLC-supply = <&reg_audio>;
+		reset-gpio = <&pca9557_b 1 1>;
+		power-domains = <&pd_mclk_out0>;
+		assigned-clocks = <&clk IMX8QXP_AUD_PLL0_DIV>,
+				<&clk IMX8QXP_AUD_ACM_AUD_REC_CLK0_DIV>,
+				<&clk IMX8QXP_AUD_MCLKOUT0>;
+		assigned-clock-rates = <786432000>, <12288000>, <12288000>;
+		status = "okay";
+	};
 };
 
 &i2c1 {
-- 
1.7.9.5

